src/main/java/com/example/bookshop/controller/BookController.java

```diff
@@ -0,0 +1,41 @@
+package com.example.bookshop.controller;
+
+import com.example.bookshop.dto.BookDto;
+import com.example.bookshop.dto.CreateBookRequestDto;
+import com.example.bookshop.service.BookService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping("/by-author")

```

Comment:
no need to specify such path, since your url will be:
```
/books/by-author?author=some_author
```

instead you should remove that suffix so that your url will be:
```
/books?author=some_author
```